### PR TITLE
[Homework #17] - class-and-functional-components by Abraham Trenyan

### PR DESCRIPTION
**Pull request by AbrahamTrenyan**
Hi Vitali! Let's talk if there's something to improve!

### CODE REVIEW

File: homeworks/abraham-trenyan/17-class-and-functional-components/src/Faces/faces.jsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
```
[the remark this answers is not in the document]
Comment (AbrahamTrenyan) [reply homeworks/abraham-trenyan/17-class-and-functional-components/src/Faces/faces.jsx]: Hey @VitalikMelnyk , just added the changes you suggested.

File: homeworks/abraham-trenyan/17-class-and-functional-components/src/Components/Faces/Faces.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import RefreshIcon from '../RefreshIcon/RefreshIcon';
+import './faces.css';
+class Faces extends Component {
+  constructor() {
+    super();
+    this.state = { images: [] };
+  }
+  async getImg(e, imgAmount) {
```
Comment (VitalikMelnyk) [inline homeworks/abraham-trenyan/17-class-and-functional-components/src/Components/Faces/Faces.jsx:9]: In react, it is bad practice to create a function that manages by a CSS class name.
Please, remake this logic by react way.
Please, review https://reactjs.org/docs/handling-events.html and docs at all.

Comment (AbrahamTrenyan) [reply homeworks/abraham-trenyan/17-class-and-functional-components/src/Components/Faces/Faces.jsx]: Hey, I changed it so know each component has its function to set the state. I hope it's OK.

**Comment by VitalikMelnyk** [comment]
Looks good @AbrahamTrenyan.
In the future, if you use async await, try to follow this approach, without combining with then/catch.
It is bad practice.
Either is fine, but mixing them is just weird. You can rewrite your code snippet in the following 2 ways:
```
function getParams(assetId) {
  const reqUrl = `https://somelink/${assetId}`
  return fetch(reqUrl)
    .then(res => res.json())
    .then(json => json.asset.params)
}
Or:

async function getParams(assetId) {
  const reqUrl = `https://somelink/${assetId}`
  const res = await fetch(reqUrl);
  const json = await res.json();
  return json.asset.params;
}
```

**Review by VitalikMelnyk — APPROVED** [review]

**Pull request merged**